protos/api/auth_service/v1/auth_service.proto

```diff
@@ -327,3 +362,119 @@ message QueueMessage {
   uint64 sequence_number = 1;
   common.v1.Ciphertext ciphertext = 2;
 }
+
+// User Handles API common messsages
+
+message UserHandleHash {
+  bytes bytes = 1;
+}
+
+// create handle
+
+message CreateHandleRequest {
+  CreateHandlePayload payload = 1;
+  common.v1.Signature handle_signature = 2;
+}
+
+message CreateHandlePayload {
+  string plaintext_handle = 1;
+  HandleVerifyingKey verifying_key = 2;
+}
+
+message HandleVerifyingKey {
+  bytes bytes = 1;
+}
+
+message CreateHandleResponse {}
+
+// delete handle
+
+message DeleteHandleRequest {
+  DeleteHandlePayload payload = 1;
+  common.v1.Signature handle_signature = 2;
+}
+
+message DeleteHandlePayload {
+  UserHandleHash hash = 1;
+}
+
+message DeleteHandleResponse {}
+
+// refresh handle
+
+message RefreshHandleRequest {
+  RefreshHandlePayload payload = 1;
+  common.v1.Signature handle_signature = 2;
+}
+
+message RefreshHandlePayload {
+  UserHandleHash hash = 1;
+}
+
+message RefreshHandleResponse {}
+
+// connect
+
+message ConnectRequest {
+  oneof step {
+    FetchConnectionPackageStep fetch = 1;
+    EnqueuePackageStep enqueue = 2;
+  }
+}
+
+message FetchConnectionPackageStep {
+  UserHandleHash hash = 1;
+}
+
+message FetchConnectionPackageResponse {
+  ConnectionPackage connection_package = 1;
+}
+
+message EnqueuePackageStep {
+  EncryptedConnectionEstablishmentPackage connection_establishment_package = 2;
```

Comment:
This is not really in scope for this PR, but we should rename this monstrosity

Reply:
Agree. Any proposal for a name?

Reply:
How about `EncryptedConnectionRequest`?

Reply:
`Request` is slightly overused and usually has a connotation of being done just in this moment.

Maybe `EncryptedConnectionPackage`? We would have a `KeyPackage` containing key material and more, and a package to "establish a connection". WDYT?

Reply:
We already have a `ConnectionPackage` (which is what the initiator picks up from the AS). And (confusingly) that's not the same as a `ConnectionEstablishmentPackage` (which is what the initiator sends back to the AS to enqueue). So calling an encrypted `ConnectionEstablishmentPackage` `EncryptedConnectionPackage` would be even more confusing IMO.

Reply:
Ah, you are right. I missed that we also have a `ConnectionPackage`.

Reply:
How about `ConnectionRequestPackage`? @kkohbrok @raphaelrobert

Reply:
How about something even shorter, like `ConnectionOffer`?